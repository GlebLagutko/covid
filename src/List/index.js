import React, {useEffect, useState} from "react";


import ListItem from "../ListItem";
import SearchBar from "../SearchBar";


function List() {
  const [countryMap, setCountryMap] = useState(new Map());
  let [input, setInput] = useState('');
  let [mapOfCountriesDefault, setMapDefault] = useState(new Map());

  function fetchData() {
    fetch('https://disease.sh/v3/covid-19/countries').then(response => response.json()).then(data => {
      const map = new Map();
      data.forEach(elem => map.set(elem.country, elem.cases));
      setMapDefault(map);
      setCountryMap(map);
    });
  }

  useEffect(() => {
    fetchData();
  }, []);

  const updateInput = (input) => {
    const filteredKeys = Array.from(mapOfCountriesDefault.keys()).filter(country => {
      return country.toLowerCase().includes(input.toLowerCase())
    });

    const currentMap = new Map();
    for (const key of mapOfCountriesDefault.keys()) {
      if (filteredKeys.includes(key))
        currentMap.set(key, mapOfCountriesDefault.get(key))
    }
    setInput(input);
    setCountryMap(currentMap);
  };

  const elems = [];
  for (const key of countryMap.keys()) {
    elems.push(<ListItem country={key} count={mapOfCountriesDefault.get(key)} key={key}/>)
  }

  return <div className='list-div '>
    <SearchBar
      keyword={input}
      setKeyword={updateInput}
    />
    <ul className='list'>

      {
        elems.map(el => el)
      }
    </ul>
  </div>
}

export default List;