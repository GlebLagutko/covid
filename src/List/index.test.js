import {render} from '@testing-library/react';

import List from '.';
import React from "react";
import {rest} from 'msw'
import {setupServer} from 'msw/node'

describe('List', () => {

  const fakeUserResponse =[{country: 'US', cases: 1000}];
  const server = setupServer(
    rest.get('https://disease.sh/v3/covid-19/countries', (req, res, ctx) => {
      return res(ctx.json(fakeUserResponse))
    }),
  );

  beforeAll(() => server.listen());
  afterAll(() => server.close());
  afterEach(() => server.resetHandlers())

  it('renders correctly', async () => {

    const {container, findByText} = render(<List/>);

    expect(await findByText("1000")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});