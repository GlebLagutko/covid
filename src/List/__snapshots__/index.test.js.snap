// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <div
    class="list-div "
  >
    <input
      placeholder="search country"
      value=""
    />
    <ul
      class="list"
    >
      <li
        class="list-item"
      >
        US
        : 
        <span>
          1000
        </span>
      </li>
    </ul>
  </div>
</div>
`;
